.github/workflows/codeql.yml

```diff
@@ -0,0 +1,58 @@
+name: CodeQL
+
+on:
+  pull_request:
+    branches:
+      - main
+      - release/*
+      - v0.*
+      - v1.*
+      - benchmarks
+  push:
+    branches:
+      - main
+      - release/*
+      - v0.*
+      - v1.*
+      - benchmarks
+  schedule:
+    - cron: "29 13 * * 2"  # weekly at 13:29 UTC on Tuesday
+
+permissions:
+  contents: read
+
+jobs:
+  analyze:
+    permissions:
+      contents: read
```

Comment:
Redundant?
```suggestion
```

Reply:
my understanding is

* technically the non-presence here removes a workflow-level permission (as opposed to being additive)
* but that since our repos are public, you can never take away `contents: read`

so I think I can remove them, let me try this in instrumentation repo first also...

Reply:
Thanks for trying.